pkg/gds/client/profiles.go

```diff
@@ -0,0 +1,179 @@
+package client
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/shibukawa/configdir"
+	"github.com/urfave/cli/v2"
+	"gopkg.in/yaml.v2"
+)
+
+// Profiles are stored in a system-specific configuration location e.g.
+// ~/Library/Application Support/rotational/gds on OS X or ~/.config/rotational/gds on
+// Linux. The profiles make it easy to switch between client configurations to connect
+// to trisatest.net or vaspdirectory.net. The profiles have a user-supplied name for
+// easy configuration and the profiles are populated with reasonable defaults.
+//
+// At most one profile can be marked as "active", this profile is treated as the default
+// profile if a specific profile is not used. If multiple profiles are marked as active
+// the "first" profile marked as active is used (with no guaranteed ordering).
+type Profiles map[string]*Profile
+
+var cfgd = configdir.New("rotational", "gds")
+
+const (
+	profileYAML = "profiles.yaml"
+)
+
+// ProfilesPath returns the location on disk where the profiles are stored. If no
+// profiles are located then an error is returned.
+func ProfilesPath() (string, error) {
+	folder := cfgd.QueryFolderContainsFile(profileYAML)
+	if folder != nil {
+		return filepath.Join(folder.Path, profileYAML), nil
+	}
+	return "", errors.New("no profiles are available")
+}
+
+// Load the profiles from disk if they're available.
+func Load() (p Profiles, err error) {
+	folder := cfgd.QueryFolderContainsFile(profileYAML)
+	if folder != nil {
+		var data []byte
+		if data, err = folder.ReadFile(profileYAML); err != nil {
+			return nil, fmt.Errorf("could not read %s: %s", profileYAML, err)
+		}
+
+		p = make(Profiles)
+		if err = yaml.Unmarshal(data, &p); err != nil {
+			return nil, fmt.Errorf("could not unmarshal profiles: %s", err)
+		}
+
+		return p, nil
+	}
+	return nil, errors.New("no profiles are available")
+}
+
+// LoadActive is a shorthand for Load() then Active() and finally Update()
+func LoadActive(c *cli.Context) (p *Profile, err error) {
+	var profiles Profiles
```

Comment:
I can see myself getting tripped up with the similarity between `Profile` and `Profiles`, maybe `Profiles` should be renamed to something like `ProfileMap`, `ProfileManifest`, or `ProfileManager`? Same goes for the filenames.